go.mod

```diff
@@ -27,10 +27,10 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/throttled/throttled/v2 v2.12.0
 	github.com/zenizh/go-capturer v0.0.0-20211219060012-52ea6c8fed04
-	go.opentelemetry.io/otel v1.19.0
+	go.opentelemetry.io/otel v1.21.0
 	go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc v0.42.0
 	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.19.0
-	go.opentelemetry.io/otel/metric v1.19.0
+	go.opentelemetry.io/otel/metric v1.21.0
```

Comment:
@lvrach one reason why this wouldn't work with dependabot is that the exporters might need updating as well (see one line above this one, the `otlptracegrpc` was still at `v1.19.0`). another reason is that we had the otel version hard-coded in our tests but I solved it in [this PR.](https://github.com/rudderlabs/rudder-go-kit/pull/219)